### PR TITLE
[Mobile] Update content hash item limit to prevent Watchman crawl failed errors

### PR DESCRIPTION
**Pull request by chipsnyder**
`Related-PR` https://github.com/wordpress-mobile/gutenberg-mobile/pull/2629

## Description

While running the metro server locally we'll see the error:

```
Loading dependency graph...jest-haste-map: Watchman crawl failed. Retrying once with node crawler.
  Usually this happens when watchman isn't running. Create an empty `.watchmanconfig` file in your project's root folder or initialize a git or hg repository in your project.
  Error: Watchman error: too many pending cache jobs. Make sure watchman is running for this project. See https://facebook.github.io/watchman/docs/troubleshooting.html.
```

This error will occur when our file size has outgrown the default cache size of metro (more info in this [SO answer](https://stackoverflow.com/a/57448116/1350218)) This increases the content hash size limit to prevent the error.

## How has this been tested?

1. Run `npm run clean`
1. Run `npm start --reset-cache`

**Expect** The Metro server to start without displaying the cache failure error

## Types of changes

Enhancement

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/react-native-editor/.watchmanconfig [hunk 1]
```diff
@@ -1 +1,3 @@
-{}
+{
+	"content_hash_max_items": 400000
```
Comment (chipsnyder) [inline packages/react-native-editor/.watchmanconfig:2]: This number was obtained by running `find . | wc -l` from the gutenberg-mobile root

**Comment by chipsnyder** [comment]
@pinarol @hypest CC'd you two on this because I'm not sure who would be the best for tooling reviews.

**Review by hypest — APPROVED** [review]
👋 Chip!

> Watchman crawl failed

I don't see that error locally using Metro from the gb-mobile demo app. Tried gb-mobile `develop` on 4559ca25.

I do remember seeing that error some times while _bundling_ the JS code for WPAndroid though. Can you confirm it happens for you during normal Metro+demo app run?

Otherwise, diff looks OK to me 👍.

**Comment by chipsnyder** [comment]
Hey @hypest Thanks for checking it for me. 

> Can you confirm it happens for you during normal Metro+demo app run?

Yeah this happens for me pretty regualrly. I will say once it clears up the error goes away for a bit but I can get it back 100% of the time if I run `npm run clean` then `npm start --reset-cache`

**Comment by hypest** [comment]
> How has this been tested?
>
>    Run npm run clean
>    Run npm start --reset-cache

Hey Chip, are those perhaps meant to be run on a parent gb-mobile checkout?

FWIW, I instead tried `npm run native clean` and `npm run native start:reset` to clean and start Metro. Thing is I can't repro the issue if I lower the setting, [similar to my comment on the gb-mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2629#issuecomment-694916497).

**Comment by chipsnyder** [comment]
>Hey Chip, are those perhaps meant to be run on a parent gb-mobile checkout?

Yeah, I was running these from the parent. 

I'm still not 100% clear on how this all works outside of it seems to work for me 😅. 

>When watching a root, if a valid JSON file named .watchmanconfig is present in the root directory, watchman will load it and use it as a source of configuration information specific to that root.
https://facebook.github.io/watchman/docs/config.html

It seems like as watchman crawls if it encounters a config file it will begin applying that config to paths below that point in the tree. So I would imagine GB could be slightly less than GB-Mobile as GB is lower in the tree but IDK for sure.

**Comment by hypest** [comment]
Hey Chip, I assigned the PR back to you just to coordinate with the gb-mobile PR after that one gets updated to resolve conflicts. From my side it's good to merge otherwise 👍.

**Pull request merged**